charts/redpanda/service_internal.go

```diff
@@ -0,0 +1,141 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// +gotohelm:filename=service.internal.go.tpl
+package redpanda
+
+import (
+	"github.com/redpanda-data/helm-charts/pkg/gotohelm/helmette"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func MonitoringEnabledLabel(dot *helmette.Dot) map[string]string {
+	values := helmette.Unwrap[Values](dot.Values)
+	if values.Monitoring.Enabled {
+		return map[string]string{
+			// no gotohelm support for strconv.FormatBool
+			"monitoring.redpanda.com/enabled": "true",
+		}
+	}
+	return map[string]string{
+		"monitoring.redpanda.com/enabled": "false",
+	}
+}
+
+func ServiceInternal(dot *helmette.Dot) *corev1.Service {
+	values := helmette.Unwrap[Values](dot.Values)
+
+	ports := []corev1.ServicePort{}
+	if values.Listeners.Admin != nil {
+		ports = append(ports, corev1.ServicePort{
+			Name:       "admin",
+			Protocol:   "TCP",
+			Port:       values.Listeners.Admin.Port,
+			TargetPort: intstr.FromInt32(values.Listeners.Admin.Port),
+		})
+	}
+	if values.Listeners.HTTP != nil && values.Listeners.HTTP.Enabled {
+		ports = append(ports, corev1.ServicePort{
+			Name:       "http",
+			Protocol:   "TCP",
+			Port:       values.Listeners.HTTP.Port,
+			TargetPort: intstr.FromInt32(values.Listeners.HTTP.Port),
+		})
+	}
+	if values.Listeners.Kafka != nil {
+		ports = append(ports, corev1.ServicePort{
+			Name:       "kafka",
+			Protocol:   "TCP",
+			Port:       values.Listeners.Kafka.Port,
+			TargetPort: intstr.FromInt32(values.Listeners.Kafka.Port),
+		})
+	}
+	ports = append(ports, corev1.ServicePort{
+		Name:       "rpc",
+		Protocol:   "TCP",
+		Port:       int32(values.Listeners.RPC.Port),
+		TargetPort: intstr.FromInt32(int32(values.Listeners.RPC.Port)),
+	})
+	if values.Listeners.SchemaRegistry != nil && values.Listeners.SchemaRegistry.Enabled {
+		ports = append(ports, corev1.ServicePort{
+			Name:       "schemaregistry",
+			Protocol:   "TCP",
+			Port:       values.Listeners.SchemaRegistry.Port,
+			TargetPort: intstr.FromInt32(values.Listeners.SchemaRegistry.Port),
+		})
+	}
+
+	annotations := map[string]string{}
+	if values.Service != nil {
+		annotations = values.Service.Internal.Annotations
+	}
+
+	return &corev1.Service{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Service",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        ServiceName(dot),
+			Namespace:   dot.Release.Namespace,
+			Labels:      helmette.Merge(FullLabels(dot), MonitoringEnabledLabel(dot)),
+			Annotations: annotations,
+		},
+		Spec: corev1.ServiceSpec{
+			Type:                     corev1.ServiceTypeClusterIP,
+			PublishNotReadyAddresses: true,
+			ClusterIP:                corev1.ClusterIPNone,
+			Selector:                 StatefulSetPodLabelsSelector(dot),
+			Ports:                    ports,
+		},
+	}
+}
+
+var _ = `
```

Comment:
Is this left over? Can we copy the comment onto `ServiceInternal`?

Reply:
Yeah, I'll clean this up. This is what old peopple do when they forget about split panes in their IDE